cognee/tasks/repo_processor/get_repo_file_dependencies.py

```
@@ -125,19 +153,38 @@ async def get_repo_file_dependencies(
-        ]
+        tasks = []
+        for file_path, lang in source_code_files[start_range : end_range + 1]:
+            # For now, only Python is supported; extend with other languages
+            if lang == 'python':
+                tasks.append(get_local_script_dependencies(repo_path, file_path, detailed_extraction))
+            else:
+                # Placeholder: create a minimal CodeFile for other languages
+                from cognee.shared.CodeGraphEntities import CodeFile
+                import aiofiles
+                async def make_codefile_stub():
+                    async with aiofiles.open(file_path, "r", encoding="utf-8") as f:
```

Ruff (0.12.2)

Function definition does not bind loop variable `file_path`

(B023)